src/allmydata/node.py

```diff
@@ -233,13 +334,11 @@ class Node(service.MultiService):
     CERTFILE = "node.pem"
     GENERATED_FILES = []
 
-    def __init__(self, config, basedir=u"."):
+    def __init__(self, config):
         service.MultiService.__init__(self)
-        # ideally, this would only be in _Config (or otherwise abstracted)
-        self.basedir = abspath_expanduser_unicode(unicode(basedir))
         # XXX don't write files in ctor!
-        fileutil.make_dirs(os.path.join(self.basedir, "private"), 0700)
-        with open(os.path.join(self.basedir, "private", "README"), "w") as f:
+        fileutil.make_dirs(os.path.join(config._basedir, "private"), 0700)
+        with open(os.path.join(config._basedir, "private", "README"), "w") as f:
             f.write(PRIV_README)
```

Comment:
Maybe this should now be something else - to avoid the use of `config._basedir` here (which appears private and so presumably foreigners like `Node` should keep their hands off).

Perhaps something based on `_Config.write_private_config` or `_Config.get_private_path` would make sense?

Reply:
I pulled in more code from the "other" branch so this stuff ends up in `write_private_config` (the `make_dirs` part) or in the `create_*` methods -- so, yay! No more creating directories in a constructor.